src/CppUTest/MemoryLeakDetector.cpp

```diff
@@ -66,7 +66,7 @@ void SimpleStringBuffer::addMemoryDump(const void* memory, size_t memorySize)
 	size_t p;
 
     while (currentPos < memorySize) {
-        add("    %04lx: ", currentPos);
+        add("    %04hx: ", (unsigned short) currentPos);
```

Comment:
Assuming that memory dumps will always be under 64k is a bit risky, isn't it?

IMHO a selection if the actual format by means of a "if(sizeof(unsigned long) == 4)" is preferable.